ingestor-default/src/main/scala/za/co/absa/hyperdrive/ingestor/implementation/writer/parquet/AbstractParquetStreamWriter.scala

```diff
@@ -0,0 +1,108 @@
+
+import scala.util.{Failure, Success, Try}
+
+private[writer] abstract class AbstractParquetStreamWriter(destination: String, val extraConfOptions: Option[Map[String, String]]) extends StreamWriter(destination) {
+
+  if (StringUtils.isBlank(destination)) {
+    throw new IllegalArgumentException(s"Invalid PARQUET destination: '$destination'")
+  }
+
+  def write(dataFrame: DataFrame, offsetManager: OffsetManager): StreamingQuery = {
+    if (dataFrame == null) {
+      throw new IllegalArgumentException("Null DataFrame.")
+    }
+
+    if (offsetManager == null) {
+      throw new IllegalArgumentException("Null OffsetManager instance.")
+    }
+
+    val outStream = getOutStream(dataFrame)
+
+    val streamWithOptions = addOptions(outStream, extraConfOptions)
+
+    val streamWithOptionsAndOffset = configureOffsets(streamWithOptions, offsetManager, dataFrame.sparkSession.sparkContext.hadoopConfiguration)
+
+    streamWithOptionsAndOffset.start(destination)
+  }
+
+  protected def getOutStream(dataFrame: DataFrame): DataStreamWriter[Row] = {
+    dataFrame
+      .writeStream
+      .trigger(Trigger.Once())
+      .format(source = "parquet")
+      .outputMode(OutputMode.Append())
+  }
+
+  protected def addOptions(outStream: DataStreamWriter[Row], extraConfOptions: Option[Map[String, String]]): DataStreamWriter[Row] = {
+    extraConfOptions match {
+      case Some(options) => options.foldLeft(outStream) {
+        case (previousOutStream, (optionKey, optionValue)) => previousOutStream.option(optionKey, optionValue)
+      }
+      case None => outStream
+    }
+  }
+
+  protected def configureOffsets(outStream: DataStreamWriter[Row], offsetManager: OffsetManager, configuration: org.apache.hadoop.conf.Configuration): DataStreamWriter[Row] = offsetManager.configureOffsets(outStream, configuration)
+}
+
+object AbstractParquetStreamWriter {
+
+  def getDestinationDirectory(configuration: Configuration): String = getOrThrow(KEY_DESTINATION_DIRECTORY, configuration, errorMessage = s"Destination directory not found. Is '$KEY_DESTINATION_DIRECTORY' defined?")
+
+  def getExtraOptions(configuration: Configuration): Option[Map[String, String]] = {
+    import scala.collection.JavaConverters._
+    val extraOptions = configuration.getKeys(KEY_EXTRA_CONFS_ROOT)
+      .asScala
+      .map(key => getKeyValueConf(key, configuration))
+
+    if (extraOptions.nonEmpty) {
+      Some(extraOptions.toMap)
+    } else {
+      None
+    }
+  }
+
+  private def getKeyValueConf(key: String, configuration: Configuration): (String, String) = {
+    Try(parseConf(configuration.getString(key.toString))) match {
+      case Success(keyValue) => keyValue
+      case Failure(exception) => throw new IllegalArgumentException(s"Invalid extra configuration for stream writer: '$key'", exception)
+    }
+  }
+
+  private def parseConf(option: String): (String, String) = {
+    val keyValue = option.split("=")
+    if (keyValue.length == 2) {
```

Comment:
I think general java conf files are allowed to have more than one `=` but only first acts as the delimiter.

Reply:
We provide the following config property: `writer.parquet.extra.conf.1=key1=value1`
That's why we check for the second =
We could change it to `writer.parquet.extra.conf.key1=value1`

However I would not change it in this PR, because `AbstractParquetStreamWriter` was just copied from `ParquetStreamWriter`

Reply:
I have created #54